### ISSUE TITLE
Could we reduce morestack when call `NewStream` in `http2Client` ?

### ISSUE THREAD

**Issue by zhexuany** [grpc/grpc-go#1797]
Please answer these questions before submitting your issue.

### What version of gRPC are you using?
1.4.2
### What version of Go are you using (`go version`)?
go version go1.9.2 darwin/amd64
### What operating system (Linux, Windows, …) and version?
Linux
### What did you do?
If possible, provide a recipe for reproducing the error.
It is not a error but a optimization opportunity. 
### What did you expect to see?
When we call `NewStream` in http2client, there should be no `morestack`
### What did you see instead?
It is there when we use perf to build a flame graph. 

engineer from Cockroach had already created a issue at `go`. You can find details via https://github.com/golang/go/issues/18138. 

In a nutshell, they propose two way to solve issue:
1. hack go and change init stack frame size. 
2. call `growStack` right before the function. 

I suggest to use second way. And idea, comments, and advice are truly welcomed.

**Comment by dfawley** [comment]
@zhexyany, can you quantify the performance penalty you're seeing?  That would help us prioritize this against other performance changes we are making.  Thanks!

**Comment by dfawley** [comment]
@zhexuany sorry, I botched the @mention above...

**Comment by zhexuany** [comment]
We already upgrade to latest grpc and high throughout/ performance has been proven. Closing this issue since it is not relevant anymore. Thanks again !!!
